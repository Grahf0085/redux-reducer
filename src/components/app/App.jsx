/* eslint-disable max-len */
import React from 'react';
import { useDispatch, useSelector } from '../../state/ReduxProvider';
import { UNDO, REDO, HISTORY } from '../../state/actions';

function App() {

  const dispatch = useDispatch();
  const selector = useSelector();

  return (
    <>
      <button onClick={() => dispatch({ type: UNDO })}>undo</button>
      <button onClick={() => dispatch({ type: REDO })}>redo</button>
      <input data-testid="in"
        type="color"
        value={selector.current}
        onChange={({ target }) => dispatch(({ type: HISTORY, payload: target.value }))}
      />
      <div data-testid="divi"
        style={{ backgroundColor: selector.current, width: '10rem', height: '10rem' }}
      ></div>
    </>
  );
}

export default App;
