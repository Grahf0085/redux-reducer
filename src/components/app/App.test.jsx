import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  
  afterEach(() => cleanup());
  
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('behavioral test', () => {
    
    render (<App />);
    
    const display = screen.getByTestId('divi');

    const input = screen.getByTestId('in');

    fireEvent.change(input, { target: { value: '#3083DC' } });
    fireEvent.change(input, { target: { value: '#7DDE92' } });
    expect(display).toHaveStyle({ backgroundColor: '#7DDE92' });

    const undo = screen.getByText('undo');
    fireEvent.click(undo);
    expect(display).toHaveStyle({ backgroundColor: '#3083DC' });

    const redo = screen.getByText('redo');
    fireEvent.click(redo);
    expect(display).toHaveStyle({ backgroundColor: '#7DDE92' });
  });

});
