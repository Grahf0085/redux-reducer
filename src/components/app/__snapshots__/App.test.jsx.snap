// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button>
    undo
  </button>
  <button>
    redo
  </button>
  <input
    data-testid="in"
    type="color"
    value="#FF0000"
  />
  <div
    data-testid="divi"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
